core/src/test/java/io/atomix/core/RaftRolesTest.java

```diff
@@ -0,0 +1,421 @@
+package io.atomix.core;
```

Comment:
Any reason why Raft tests are not in the raft module?

Reply:
Most of the tests was added before. I just moved them. Here they are more integration tests, which I would prefer here. I think they are also written here because the setup was easier.